apps/tangle-dapp/data/DelegationsPayouts/usePayouts.ts

```diff
@@ -0,0 +1,288 @@
+                        typeof rewards.individual === 'object' &&
+                        rewards.individual !== null
+                      ) {
+                        Object.entries(rewards.individual).forEach(
+                          ([key, value]) => {
+                            if (key === validator.toString()) {
+                              validatorRewardPoints = Number(value);
+                            }
+                          }
+                        );
+                      }
+
+                      if (validatorRewardPoints > 0) {
+                        allRewards.push({
+                          era,
+                          totalRewardPoints,
+                          validator: validator.toString(),
+                          validatorRewardPoints,
+                        });
+                      }
+                    }
+                  }
+                });
+
+                validatorPayoutsPromises = allRewards
+                  .map(async (reward) => {
+                    const {
+                      era,
+                      totalRewardPoints,
+                      validator,
+                      validatorRewardPoints,
+                    } = reward;
+                    const validatorLedger =
+                      await apiPromise.query.staking.ledger(validator);
+
+                    const claimedEras = validatorLedger
+                      .unwrap()
+                      .claimedRewards.map((era) =>
+                        Number(era.toString().replace(/,/g, ''))
+                      );
+
+                    if (claimedEras.includes(era)) return;
+
+                    const erasTotalReward =
+                      await apiPromise.query.staking.erasValidatorReward(era);
+
+                    if (erasTotalReward.isSome) {
+                      const validatorTotalReward =
+                        (validatorRewardPoints *
+                          Number(erasTotalReward.unwrap().toString())) /
+                        totalRewardPoints;
+
+                      if (validatorTotalReward > 0) {
+                        const validatorTotalRewardFormatted =
+                          await formatTokenBalance(
+                            new u128(
+                              apiPromise.registry,
+                              validatorTotalReward.toString()
+                            )
+                          );
+
+                        const eraStaker =
+                          await apiPromise.query.staking.erasStakers(
+                            era,
+                            validator
+                          );
+
+                        const validatorTotalStake = eraStaker.total.unwrap();
+
+                        const validatorTotalStakeFormatted =
+                          await formatTokenBalance(validatorTotalStake);
+
+                        if (
+                          Number(validatorTotalStake.toString()) > 0 &&
```

Comment:
This conversion may fail if `Number` can't hold larger Big Integer (`BN`)? If it is a `BN`, it should provide you with functions to accomplish comparisons, ex. `validatorTotalStake.gt(new BN(0))`